### PR TITLE
feat: use document title as root node (optionally)

### PR DESCRIPTION
**Pull request by james-tindal**
Implements #60.

With one top-level heading:

With multiple top-level headings:

### CODE REVIEW

File: src/mindmap-view.ts [hunk 1]
```diff
@@ -265,10 +265,15 @@ export default class MindmapView extends ItemView {
   async transformMarkdown(markdown: string) {
     let { root, features, frontmatter } = this.transformer.transform(markdown);
 
+    const rootWithTitle =
```
Comment (MasterTuto) [inline src/mindmap-view.ts:268]: - make a new method to that takes root and returns a new root with title, this new method must be named as camelCase.
- Replace the multiple ternary operators with `if` clauses

**Review by MasterTuto — CHANGES_REQUESTED** [review]
Thanks for leaving your PR, I'd be glad if you made the changes I have requests.

Thanks for your contribution.

**Comment by AdrianSimionov** [comment]
Can we please add documentation and a front-matter option on top of the plugin settings? The front-matter option will take precedence and be applied only to the current .md file.

**Comment by james-tindal** [comment]
I'll work on the front matter issues tomorrow morning and write it up in the readme

**Review by MasterTuto — APPROVED** [review]

**Comment by james-tindal** [comment]
@AdrianSimionov I followed the image, resolved the conflicts, merged and committed, but I don't have permission to push from the CLI.

**Comment by james-tindal** [comment]
I can merge it differently though so that shouldn't be a problem.

**Comment by james-tindal** [comment]
I noticed the map doesn't update when the setting changes. PR #86 (multiple tabs) sets up some stuff that I will base this functionality on, so I'm waiting on that.

**Pull request merged**